src/kohlrahbi/ahbtable/ahbcondtions.py

```diff
@@ -0,0 +1,115 @@
+
+    @classmethod
+    def from_docx_table(cls, docx_tables: list[DocxTable], edifact_format: EdifactFormat) -> "AhbConditions":
+        """
+        Create an AhbPackageTable object from a docx table.
+
+        """
+        table_data = []
+        for table in docx_tables:
+            for row in table.rows:
+                if row.cells[-1].text and row.cells[0].text != "EDIFACT Struktur":
+                    row_data = row.cells[-1].text
+                    table_data.append(row_data)
+
+        conditions_dict = {}
+        are_there_conditions = len(table_data) > 0
+        if are_there_conditions:
+            conditions_dict = AhbConditions.collect_conditions(
+                conditions_list=table_data, edifact_format=edifact_format
+            )
+
+        return cls(conditions_dict=conditions_dict)
+
+    @staticmethod
+    def collect_conditions(
+        conditions_list: list[str], edifact_format: EdifactFormat
+    ) -> dict[EdifactFormat, dict[str, str]]:
+        """collect conditions from list of all conditions and store them in conditions dict."""
+        conditions_dict: dict[EdifactFormat, dict[str, str]] = {edifact_format: {}}
+
+        conditions_str = "".join(conditions_list)
+
+        # Split the input into parts enclosed in square brackets and other parts
+        matches = re.findall(
+            r"\[(\d+)](.*?)(?=\[\d+]|$)",
+            conditions_str,
+            re.DOTALL,
+        )
+        for match in matches:
+            # make text prettier:
+            text = match[1].strip()
+            text = re.sub(r"\s+", " ", text)
+            # check whether condition was already collected:
+            existing_text = conditions_dict[edifact_format].get(match[0])
+            is_condition_key_collected_yet = existing_text is not None
+            if is_condition_key_collected_yet:
+                key_exits_but_shorter_text = len(text) > len(existing_text)  # type: ignore[arg-type]
+            if not is_condition_key_collected_yet or key_exits_but_shorter_text:
+                conditions_dict[edifact_format][match[0]] = text
+
+        logger.info("The package conditions for %s were collected.", edifact_format)
+        return conditions_dict
+
+    def include_condition_dict(self, to_add=dict[EdifactFormat, dict[str, str]] | None) -> None:
+        """ " Include a dict of conditions to the conditions_dict"""
+        if to_add is None:
+            logger.info("Conditions dict to be added is empty.")
+        for edifact_format, edi_cond_dict in to_add.items():
+            for condition_key, condition_text in edi_cond_dict.items():
+                if edifact_format in self.conditions_dict:
+                    if condition_key in self.conditions_dict[edifact_format]:
+                        if len(condition_text) > len(self.conditions_dict[edifact_format][condition_key]):
+                            self.conditions_dict[edifact_format][condition_key] = condition_text
```

Comment:
I guess we can combine the two conditions here don't we?
```suggestion
                    if condition_key in self.conditions_dict[edifact_format] and len(condition_text) > len(self.conditions_dict[edifact_format][condition_key]):
                        self.conditions_dict[edifact_format][condition_key] = condition_text
```